Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM --platform=linux/amd64 python:3.9
+
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+ENV PYTHONPATH /
+
+# Get necessary system packages
+RUN apt-get update \
+  && apt-get install --no-install-recommends --yes \
+     build-essential \
+     python3 \
+     python3-pip \
+     python3-dev \
+     mariadb-client \
+  && rm -rf /var/lib/apt/lists/*
+
+# Get necessary python libraries
+COPY requirements.txt .
+RUN pip3 install --compile --no-cache-dir -r requirements.txt
```

Comment:
After running this, you get an error saying your mariadb connector is outdated as well as pip needs to be upgraded. I had this exact same error so it may be my machine and how I solved this issues. If it is infact not my machine I you can try a few things.
1. use this command and it will update your connector: RUN wget https://dlm.mariadb.com/2678574/Connectors/c/connector-c-3.3.3/mariadb-connector-c-3.3.3-debian-bullseye-amd64.tar.gz -O - | tar -zxf - --strip-components=1 -C /usr

2. You can specify which verision of MariaDB you want to use in the requirements. Here you will need to use an older version that is compatible with your current connector

3. For pip, you can use this command: RUN pip install --upgrade pip. What I think the issue here is that the container is using a different version of pip and mariadb than what your local is? Honestly not sure but these were my workarounds